utils/mail.go

```diff
@@ -59,22 +60,17 @@ func newSMTPClient(conn net.Conn, config *model.Config) (*smtp.Client, *model.Ap
 		}
 	}
 
-	auth := smtp.PlainAuth("", config.EmailSettings.SMTPUsername, config.EmailSettings.SMTPPassword, config.EmailSettings.SMTPServer+":"+config.EmailSettings.SMTPPort)
-	if config.EmailSettings.ConnectionSecurity == model.CONN_SECURITY_TLS {
-		if err = c.Auth(auth); err != nil {
-			return nil, model.NewLocAppError("SendMail", "utils.mail.new_client.auth.app_error", nil, err.Error())
-		}
-	} else if config.EmailSettings.ConnectionSecurity == model.CONN_SECURITY_STARTTLS {
+	if config.EmailSettings.ConnectionSecurity == model.CONN_SECURITY_STARTTLS {
```

Comment:
@crspeller I think it makes more sense to make sure the running configuration is compatible and correct as soon as possible. If we leave it to the `mail.go` file then when the front end (or any other existing or future reference) tries to pull these values they may or may not be correct depending on whether or not `mail.go` has executed since the server got turned up. 
So, for consistency I think the `config.go` is the best location to update outdated configurations options to make sure the configuration is correct across all aspects of the platform.